src/openeo_odc/map_to_odc.py

```diff
@@ -110,15 +111,28 @@ def resolve_from_parameter(node):
     return in_nodes
 
 
-def create_job_header(dask_url: str, odc_env_collection: str = "default", odc_env_user_gen: str = "user_generated"):
+def create_job_header(dask_url: str, odc_env_collection: str = "default", odc_env_user_gen: str = "user_generated", job_id: str = "Unassigned", user_id: str = "Unassigned"):
```

Comment:
I thought including the user ID in-case we wanted to aggregate by what the user has done, this isn't the case, but I think it's worth keeping as that could potentially be some useful insight. 

Jobs without ID's would be jobs coming from us when working directly with the gateway, we can for sure throw and error, but we will always have to defined the options if we do!

What would a null user look like?

Reply:
Ah, yes, we definitely want to include `user_id`, I was just referring to what happens if `user_id` is missing from the arguments!

The reason why I raise this is the following:
If we make a mistake somewhere and forget to set the user_id for a job, this job will now look like a job that we've submitted for testing purposes and we'll most likely not spot it for a while (which could lead to undetected mistakes in e.g. the cost accounting). By using "Unassigned" locally here, we'll only catch such jobs that didn't have a user set right here, but such a mistake elsewhere would still slip through. It's also hard to maintain, because we might use "No user"/"missing"/"na" or similar strings elsewhere and suddenly you can't search for jobs without users that easily anymore. If we instead define that every job _must_ have an associated user (even if that is a mock-user for us devs to use) and simply throw exceptions when a user isn't present, we'll notice our mistakes much earlier! 
An alternative without exceptions would be a null-user. i.e. a user id that is defined somewhere central and always gets applied when there's no user id present!